### ISSUE TITLE
Rule 973337 misses word boundaries and leads to lots of false positives

### ISSUE THREAD

**Issue by xformerfhs** [SpiderLabs/owasp-modsecurity-crs#244]
Rule 973337 misses word boundaries and triggers on a base64 encoded parameter that contains "...9ONn0U8=.." as there is an "on" followed by a "=". This is obviously not correct and addind the correct word boundaries to this rule would make it work correctly.

CRS version is 2.2.9.

**Comment by dune73** [comment]
This is 941120 PL1 after the renumbering. And still open.

**Comment by lifeforms** [comment]
XSS rules are always tricky because browsers are so "helpful" in parsing and there are lots of interactions with encoding.

We should be careful about adding a word boundary `\b` in front of the `on` token. We must make sure to match something like `<x%20onclick=alert(1)`. If we just add a word boundary we may miss this case because numbers are seen as part of a word. We might also miss other creative cases.

```
<x onclick=alert(1): 1
<xblaonclick=alert(1): 0
<x%20onclick=alert(1): 0
```

What we could do, is reduce FP by requiring `[a-zA-Z]+` after `on` though, instead of `\w` as it is now. We can assume safely that `ONn0U8` will never be a valid event handler name due to the use of numbers. This should probably fix most base64 FPs.

**Comment by lifeforms** [comment]
This issue will be fixed in 3.0.1. Thanks for reporting @xformerfhs!

More discussion is in #675 where another user reported the same problem.
